File: estudos/matrix/SortReverse.java
import java.util.Arrays;
import java.util.Collections;
import java.util.Scanner;

public class SortReverse {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Enter the number of elements: ");
        int n = scanner.nextInt();
        
        Double[] array = new Double[n];
        
        System.out.println("Enter " + n + " double values:");
        for (int i = 0; i < n; i++) {
            array[i] = scanner.nextDouble();
        }
        
        // Sort the array
        // Arrays.sort(array);

        // reverse
        Arrays.sort(array, Collections.reverseOrder());
        
        // Print the sorted array
        System.out.println("Sorted array:");
        for (Double num : array) {
            System.out.print(num + " ");
        }
        
        scanner.close();
    }
}

File: estudos/matrix/sortReverse.cpp
#include <iostream>
#include <algorithm>
#include <iomanip>

using namespace std;

int main() {
    int n;
    cout << "Enter the number of elements: ";
    cin >> n;
    
    // Dynamically allocate array
    double* array = new double[n];
    
    cout << "Enter " << n << " double values:" << endl;
    for (int i = 0; i < n; i++) {
        cin >> array[i];
    }
    
    // Sort the array
    sort(array, array + n, greater<double>());
    
    // Print the sorted array
    cout << "Sorted array:" << endl;
    for (int i = 0; i < n; i++) {
        cout << array[i] << " ";
    }
    cout << endl;
    
    // Free the allocated memory
    delete[] array;
    
    return 0;
}